src/extension.ts

```diff
@@ -42,11 +46,53 @@ export interface Api {
 /**
  * Activate the extension. This is the main entry point.
  */
-export async function activate(context: vscode.ExtensionContext): Promise<Api> {
+export async function activate(context: vscode.ExtensionContext): Promise<Api | undefined> {
     try {
         console.debug("Activating Swift for Visual Studio Code...");
 
-        const workspaceContext = await WorkspaceContext.create();
+        const outputChannel = new SwiftOutputChannel();
+        const toolchain: SwiftToolchain | undefined = await SwiftToolchain.create()
+            .then(toolchain => {
+                outputChannel.log(toolchain.swiftVersionString);
+                toolchain.logDiagnostics(outputChannel);
+                contextKeys.createNewProjectAvailable = toolchain.swiftVersion.isGreaterThanOrEqual(
+                    new Version(5, 8, 0)
+                );
+                return toolchain;
+            })
+            .catch(error => {
+                outputChannel.log("Failed to discover Swift toolchain");
+                outputChannel.log(error);
+                contextKeys.createNewProjectAvailable = false;
+                return undefined;
+            });
+
+        context.subscriptions.push(...commands.registerToolchainCommands(toolchain));
+        context.subscriptions.push(
+            vscode.workspace.onDidChangeConfiguration(event => {
+                // on toolchain config change, reload window
+                if (event.affectsConfiguration("swift.path")) {
+                    showReloadExtensionNotification(
+                        "Changing the Swift path requires the project be reloaded."
```

Comment:
You may not have a project loaded, you may have multiple projects loaded...
```suggestion
                        "Changing the Swift path requires Visual Studio Code be reloaded."
```

Reply:
done